listenbrainz/webserver/static/js/src/stats/UserEntityChart.tsx

```diff
@@ -95,6 +95,8 @@ export default class UserEntityChart extends React.Component<
       event.preventDefault();
     }
 
+    if (newPage <= 0) return; // Checking if newPage in query is a positive integer.
```

Comment:
```suggestion
    const { totalPages } = this.state;
    if (newPage <= 0 || newPage > totalPages) return; // Checking if newPage in query is a positive integer.
```

That way we guard against going over the total number of pages as well (which does not break anything currently but results in empty pages).

Please also add a small test for this as you did for newPage <= 0.